scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -142,6 +142,31 @@ object FileSource {
     !glob(globPath, conf, SuccessFileFilter).isEmpty
   }
 
+  def allPathsCommitted(globPath: String, conf: Configuration): Boolean = {
+    case class PathUsage(val path: Path, committed: Boolean)
+
+    val usedDirs: Iterable[PathUsage] = glob(globPath, conf, AcceptAllPathFilter)
+      .map { fileStatus: FileStatus =>
+        val dir =
+          if (fileStatus.isDirectory)
+            fileStatus.getPath
+          else
+            fileStatus.getPath.getParent
+
+        if (fileStatus.isFile && SuccessFileFilter.accept(fileStatus.getPath)) {
+          PathUsage(dir, true)
+        } else {
+          PathUsage(dir, false)
+        }
+      }
+      .groupBy { case PathUsage(path, _) => path }
```

Comment:
You can just do (dir, OrVal(true)) (dir, OrVal(false))

and then MapAlgebra.sumByKey(listOfPaths) will do the right thing